json/src/de.rs

```diff
@@ -21,8 +21,19 @@ pub struct Deserializer<R> {
     remaining_depth: u8,
 }
 
-impl<R> Deserializer<R> {
-    fn new(read: R) -> Self {
+impl<R> Deserializer<R>
+    where R: read::Read
+{
+    /// Create a JSON deserializer from one of the possible serde_json input
+    /// sources.
+    ///
+    /// Typically it is more convenient to use one of these methods instead:
+    ///
+    ///   - Deserializer::from_str
+    ///   - Deserializer::from_bytes
+    ///   - Deserializer::from_iter
+    ///   - Deserializer::from_reader
+    pub fn new(read: R) -> Self {
```

Comment:
Please note that for my use case, exposing just the types would be enough and these methods could still be private.